packages/graphql-testing/src/test/MockApolloLink.test.ts

```diff
@@ -0,0 +1,57 @@
+import MockApolloLink from '../MockApolloLink';
+import {MockGraphQLResponse} from '../types';
+
+import petQuery from './fixtures/PetQuery.graphql';
+
+const mockRequest = {
+  query: petQuery,
+  operationName: 'Pets',
+  variables: {},
+  extensions: {},
+  setContext: () => ({}),
+  getContext: () => ({}),
+  toKey: () => '',
+};
+
+describe('MockApolloLink', () => {
+  it('returns error message with empty mock object', async () => {
+    const mockApolloLink = new MockApolloLink({});
+
+    const result = await new Promise(resolve => {
+      mockApolloLink.request(mockRequest).subscribe(resolve);
+    });
+
+    expect(result).toMatchObject({
+      message:
+        "Can’t perform GraphQL operation 'Pets' because no valid mocks were found (it looks like you tried to provide data directly to the mock GraphQL client. You need to provide your fixture on the key that matches its operation name. To fix this, simply change your code to read 'mockGraphQLClient({Pets: yourFixture}).'",
+    });
+  });
+
+  it('returns error message when there are no matching mocks', async () => {
+    const mockApolloLink = new MockApolloLink({LostPets: {}, PetsForSale: {}});
+
+    const result = await new Promise(resolve => {
+      mockApolloLink.request(mockRequest).subscribe(resolve);
```

Comment:
I think there is maybe a utility that does this work for you? You can check the Apollo link tests in web.

Reply:
do you mean this file?
should just being the entire file over and include it in `graphql-testing`? I can use it for the test of `Operations` too.

Reply:
Just this one function: https://github.com/Shopify/web/blob/master/packages/@shopify/apollo-link-test-utilities/index.ts#L38, for now I thin it would be fine to just copy that over.